compression.go

```diff
@@ -65,6 +65,9 @@ func newCompressionPool(
 	newDecompressor func() Decompressor,
 	newCompressor func() Compressor,
 ) *compressionPool {
+	if newDecompressor == nil && newCompressor == nil {
```

Comment:
Perhaps OR instead of AND?

Reply:
AND matches the test that replaces it [here](https://github.com/bufbuild/connect-go/pull/493/files#diff-086a37ef11c8fd9a2516991339db79c207741ac98dd89c8215d31a8bead2814fL51).

But I definitely see that OR is much safer, in terms of avoiding potential runtime panics. However it is likely to produce inexplicible behavior later, when the compressor they tried to register just doesn't work. One being nil and the other not seems like a clear programmer error; would it be acceptable to instead fail hard and fast: panic immediately if one is nil and the other is not?

Reply:
Hm, that's fair. I'd personally prefer a panic right here, but that's hard-banned by the internal style guide. Let's keep it as is.